lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -273,7 +283,7 @@ public function bindValue($param, $value, int $type = ParameterType::STRING) : v
      */
     public function bindParam($param, &$variable, int $type = ParameterType::STRING, ?int $length = null) : void
     {
-        $param = $this->_paramMap[$param];
+        $param = $this->_paramMap[$param] ?? (string) $param;
```

Comment:
Why is casting needed? I believe the logic should be about the following:
```php
if (is_int($param)) {
    if (! isset($this->_paramMap[$param])) {
        throw new OCI8Exception(/* ... */);
    }

    $param = $this->_paramMap[$param];
}
```

Reply:
> Why is casting needed?
Without casting to the string produces a phpstan error.


> I believe the logic should be about the following:

Ok